File: eventues-frontend/firebase.ts
// firebase.ts
import { initializeApp, getApps, FirebaseApp } from "firebase/app";
import { getAuth, Auth } from "firebase/auth";
import { envConfig } from "./lib/config/environment";

// Get Firebase configuration from environment manager
const firebaseConfig = envConfig.getFirebaseConfig();

// Inicializa o Firebase apenas uma vez
let firebaseApp: FirebaseApp;
let auth: Auth;

if (!getApps().length) {
  firebaseApp = initializeApp(firebaseConfig);
  auth = getAuth(firebaseApp);
  console.log("Firebase initialized");
} else {
  firebaseApp = getApps()[0];
  auth = getAuth(firebaseApp);
  console.log("Firebase already initialized");
}

export { auth };

File: eventues-backend/chalicelib/src/utils/firebase.py
import firebase_admin
from firebase_admin import credentials, firestore, auth, storage
from chalicelib.src.config.environment import env_config

# Get Firebase configuration for current environment
firebase_config = env_config.get_firebase_config()

# Initialize Firebase with environment-specific configuration
cred = credentials.Certificate(firebase_config['credentials'])
firebase_admin.initialize_app(cred, {
    'storageBucket': firebase_config['storage_bucket']
})

# Conexão com Firestore e Storage
db = firestore.client()
bucket = storage.bucket()

# Função para verificar tokens de autenticação do Firebase
def verify_token(token: str) -> str:
    try:
        decoded_token = auth.verify_id_token(token)
        return decoded_token["uid"]
    except firebase_admin.auth.InvalidIdTokenError:
        raise ValueError("Token de autenticação inválido.")
    except firebase_admin.auth.ExpiredIdTokenError:
        raise ValueError("Token de autenticação expirado.")
    except Exception as e:
        raise ValueError(f"Erro na verificação do token: {str(e)}")
